### ISSUE TITLE
Fix Precompilation

### ISSUE THREAD

**Issue by ysimillides** [JuliaDiffEq/FEniCS.jl#28]
Currently, we do not precompile the FEniCS.jl package. This is due to a known error documented here https://github.com/JuliaPy/PyCall.jl#using-pycall-from-julia-modules . 
The fix can be documented here https://gist.github.com/ysimillides/f7b6d01c646d1d13034ec1069eec4104 . 
To implement the fix though, would require refactoring of a large segment of the code base, so I will leave this until September.

**Comment by PallHaraldsson** [comment]
I was reading https://julialang.org/blog/2017/09/gsoc-fenics and see:

1.
"Fixing precompilation which would provide a large performance benefit."

My understanding, is precompilation helps only with startup (but not general use after). If I'm right, and that's not a big deal, then maybe amend the too pessimistic wording in the interesting blog post?

2.
"initial partial different equations"

typo missing -al, right.

**Comment by ChrisRackauckas** [comment]
>If I'm right, and that's not a big deal, then maybe amend the too pessimistic wording in the interesting blog post?

You are correct. It should be re-worded to say it would help speed up initial calls but that's it.

**Comment by ysimillides** [comment]
Hi Pall,

Thanks for the kind words. Once theres abit more of the package implemented, I'll see about updating / creating a new blog post, hopefully with a more positive tone :)

**Comment by clason** [comment]
Now that precompile is the default, it might make sense to explicitly mention this in the README, otherwise people will wonder (like I did) why FEniCS.jl keeps precompiling after every startup...
